march_hardware/src/TemperatureGES.cpp

```diff
@@ -5,17 +5,18 @@
 
 namespace march4cpp
 {
-TemperatureGES::TemperatureGES(int slaveIndex, int byteOffset) : Slave(slaveIndex)
+TemperatureGES::TemperatureGES(int slaveIndex, int temperatureByteOffset) : Slave(slaveIndex)
 {
-  ROS_ASSERT_MSG(byteOffset >= 0, "Slave configuration error: byteOffset %d can not be negative.", byteOffset);
-  this->byteOffset = byteOffset;
+  ROS_ASSERT_MSG(temperatureByteOffset >= 0, "Slave configuration error: temperatureByteOffset %d can not be negative.",
+                 temperatureByteOffset);
+  this->temperatureByteOffset = temperatureByteOffset;
 }
 
 float TemperatureGES::getTemperature()
 {
-  // TODO(Martijn) read actual data from ethercat.
-  union bit8 return_byte = get_input_bit8(static_cast<uint16>(slaveIndex), static_cast<uint8>(byteOffset));
-  return static_cast<float>(return_byte.ui);
+  union bit32 temperature =
+      get_input_bit32(static_cast<uint16>(slaveIndex), static_cast<uint8>(this->temperatureByteOffset));
+  return temperature.f;
```

Comment:
Make sure this is in normalized degrees Celsius

Reply:
Test confirmed this is the case.